File: Sample Codes/Visual Basic .Net/Polling/Polling/Polling.vb
Public Class Polling
    Dim retCode As Object
    Dim hReader As Integer
    Dim connActive As Boolean
    Dim StartPoll As Boolean

    Dim Sendbuff(0 To 262) As Byte
    Dim RecvBuff(0 To 262) As Byte
    Dim SendLen, RecvLen As Long

    Private Sub Polling_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        InitMenu()
    End Sub

    Private Sub InitMenu()
        Dim i As Integer

        For i = 1 To 10
            cbPort.Items.Add("COM" & i)
        Next i
        cbPort.SelectedIndex = 0

        lCardLabel.Text = "Card Type:"
        StartPoll = True
        bConn.Enabled = True
        connActive = False

    End Sub

    Private Sub bConn_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bConn.Click
        Dim a As String

        a = cbPort.Text

        retCode = ACR122_Open(a, hReader)

        If retCode = 0 Then

            connActive = True
            bConn.Enabled = False
            bStartPoll.Enabled = True

            MsgBox("Connection to " & a & " Success!", MsgBoxStyle.Information + MsgBoxStyle.OkOnly)

        Else
            MsgBox("Connection to " & a & " Failed!", MsgBoxStyle.Information + MsgBoxStyle.OkOnly)

        End If
    End Sub

    Private Sub bQuit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bQuit.Click
        If connActive = True Then
            ACR122_Close(hReader)
        End If

        Me.Close()
    End Sub

    Private Sub bStartPoll_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bStartPoll.Click

        If StartPoll = True Then
            bStartPoll.Text = "Stop Polling"
            StartPoll = False
            tPollTimer.Enabled = True
        Else
            bStartPoll.Text = "Start Polling"
            StartPoll = True
            tPollTimer.Enabled = False
        End If

    End Sub

    Private Sub bPollTimer_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles tPollTimer.Tick
        Dim CardType As Byte
        Dim tmpStr As String

        Sendbuff(0) = &HD4
        Sendbuff(1) = &H60
        Sendbuff(2) = &H1
        Sendbuff(3) = &H1
        Sendbuff(4) = &H20
        Sendbuff(5) = &H23
        Sendbuff(6) = &H11
        Sendbuff(7) = &H4
        Sendbuff(8) = &H10

        SendLen = 9
        RecvLen = 255

        retCode = ACR122_DirectTransmit(hReader, Sendbuff(0), SendLen, RecvBuff(0), RecvLen)

        If retCode = 0 Then
            If RecvLen > 3 Then
                CardType = RecvBuff(8)

                Select Case CardType
                    Case &H18
                        tmpStr = "Mifare 4K "
                    Case &H0
                        tmpStr = "Mifare Ultralight "
                    Case &H28
                        tmpStr = "ISO 14443-4 Type A "
                    Case &H8
                        tmpStr = "Mifare 1K "
                    Case &H9
                        tmpStr = "Mifare Mini "
                    Case &H20
                        tmpStr = "Mifare DesFire "
                    Case &H98
                        tmpStr = "Gemplus MPCOS "
                    Case Else
                        CardType = RecvBuff(3)
                        Select Case CardType
                            Case &H23
                                tmpStr = "ISO 14443-4 Type B "
                            Case &H11
                                tmpStr = "Felica 212K "
                            Case &H4
                                tmpStr = "Topaz "
                            Case Else
                                tmpStr = "Unknown card "
                        End Select
                End Select
            Else
                tmpStr = ""
            End If

            lCardType.Text = tmpStr
        Else
            lCardType.Text = "Error: " & retCode
        End If

    End Sub
End Class

File: Sample Codes/Java/source files/Polling.java
/*
  Copyright(C):      Advanced Card Systems Ltd

  File:              DeviceProgramming.java

  Description:       This sample program outlines the steps on how to
                     poll PICC cards in ACR122S

  Author:            Alain Benedict Chua

  Date:              Nov 11, 2009

  Revision Trail:   (Date/Author/Description)

======================================================================*/

import ACR122.*;

import java.awt.*;
import java.awt.event.*;

import javax.swing.*;

public class Polling extends JFrame implements ActionListener{
	
	static ACR122Loader acr122 = new ACR122Loader();
	
	int retCode;
	boolean connActive;
	boolean enablePoll;
	
	int [] hReader = new int[1]; 
	int [] RecvLen = new int[1];
	int SendLen = 0;
	byte [] SendBuff = new byte[255];
	byte [] RecvBuff = new byte[255];
	String PrintText = new String();
	String tmpData = new String();
	Timer timer;
	
	private java.awt.Button bConn;
    private java.awt.Button bQuit;
    private java.awt.Button bStart;
    private javax.swing.JComboBox cbPort;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel lCard;
    private javax.swing.JLabel lCardType;
    private javax.swing.JPanel mainPanel;
	
    public Polling() {
    	
    	this.setTitle("Other PICC Cards");
        initComponents();
        initMenu();
    }
    
    private void initComponents() {
    	
    		
    	 Image icon = Toolkit.getDefaultToolkit().getImage("ACS_multiple.PNG");
    	
       	 this.setIconImage(icon);
    	
    	 setSize(240, 241);
    	
    	//sets the location of the form at the center of screen
		Dimension dim = getToolkit().getScreenSize();
		Rectangle abounds = getBounds();
		setLocation((dim.width - abounds.width) / 2, (dim.height - abounds.height) / 2);
		requestFocus();
    		
    	 mainPanel = new javax.swing.JPanel();
         jLabel1 = new javax.swing.JLabel();
         cbPort = new javax.swing.JComboBox();
         bConn = new java.awt.Button();
         bStart = new java.awt.Button();
         bQuit = new java.awt.Button();
         lCard = new javax.swing.JLabel();
         lCardType = new javax.swing.JLabel();

         mainPanel.setName("mainPanel"); 

         
         jLabel1.setText("Select Port:"); 
         jLabel1.setName("jLabel1"); 

         cbPort.setModel(new javax.swing.DefaultComboBoxModel(new String[] { "COM1", "COM2", "COM3", "COM4", "COM5", "COM6", "COM7", "COM8", "COM9", "COM10" }));
         cbPort.setName("cbPort"); 

         bConn.setLabel("Connect"); 
         bConn.setName("bConn"); 

         bStart.setLabel("Start Polling"); 
         bStart.setName("bStart"); 

         bQuit.setLabel("Quit"); 
         bQuit.setName("bQuit"); 

         lCard.setText("Card Type:"); 
         lCard.setName("lCard"); 

         lCardType.setText(""); 
         lCardType.setName("lCardType"); 

         javax.swing.GroupLayout mainPanelLayout = new javax.swing.GroupLayout(getContentPane());
         getContentPane().setLayout(mainPanelLayout);
         mainPanelLayout.setHorizontalGroup(
             mainPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
             .addGroup(mainPanelLayout.createSequentialGroup()
                 .addContainerGap()
                 .addGroup(mainPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
                     .addGroup(mainPanelLayout.createSequentialGroup()
                         .addComponent(jLabel1)
                         .addGap(18, 18, 18)
                         .addComponent(cbPort, javax.swing.GroupLayout.PREFERRED_SIZE, 102, javax.swing.GroupLayout.PREFERRED_SIZE))
                     .addComponent(bQuit, javax.swing.GroupLayout.DEFAULT_SIZE, 210, Short.MAX_VALUE)
                     .addComponent(bStart, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                     .addGroup(mainPanelLayout.createSequentialGroup()
                         .addComponent(lCard)
                         .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                         .addComponent(lCardType, javax.swing.GroupLayout.PREFERRED_SIZE, 55, javax.swing.GroupLayout.PREFERRED_SIZE))
                     .addComponent(bConn, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
                 .addContainerGap(14, Short.MAX_VALUE))
         );
         mainPanelLayout.setVerticalGroup(
             mainPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
             .addGroup(mainPanelLayout.createSequentialGroup()
                 .addContainerGap()
                 .addGroup(mainPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                     .addComponent(jLabel1)
                     .addComponent(cbPort, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                 .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                 .addComponent(bConn, javax.swing.GroupLayout.PREFERRED_SIZE, 35, javax.swing.GroupLayout.PREFERRED_SIZE)
                 .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                 .addComponent(bStart, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                 .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                 .addComponent(bQuit, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                 .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                 .addGroup(mainPanelLayout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                     .addComponent(lCard)
                     .addComponent(lCardType))
                 .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
         );

         mainPanelLayout.linkSize(javax.swing.SwingConstants.VERTICAL, new java.awt.Component[] {bConn, bQuit, bStart});
         
         bConn.addActionListener(this);
         bStart.addActionListener(this);
         bQuit.addActionListener(this);
    }
    
    public void initMenu(){

    	bConn.setEnabled(true);
    	bStart.setEnabled(false);
    	
    	enablePoll = false;
    	
    	timer = new Timer(1000, this);
    	
    }
    
    public void actionPerformed(ActionEvent e) {
    	
    	if(bConn == e.getSource())
		{
			String port = (String)cbPort.getSelectedItem();
			
			retCode = acr122.jACR122_Open(port, hReader);
			
			if(retCode == 0)
			{
				connActive = true;
				bConn.setEnabled(false);
				bStart.setEnabled(true);
				
				JOptionPane.showMessageDialog( null, "Connection Success");
			}
			else
			{
				JOptionPane.showMessageDialog( null, "Connection Failed");
			}
			
		}
    	
    	if(bStart == e.getSource())
    	{
    		if(enablePoll == false)
    		{
    		   enablePoll = true;
    		   bStart.setLabel("Stop Polling");
    		   timer.start();
    		}
    		else
    		{
    		   enablePoll = false;
    		   bStart.setLabel("Start Polling");
    		   lCardType.setText("");
			   timer.stop();
					
    		}
    		
    	}
    	
    	if(timer == e.getSource())
    	{
    		SendBuff[0] = (byte) 0xD4;
			SendBuff[1] = (byte) 0x60;
			SendBuff[2] = (byte) 0x01;
			SendBuff[3] = (byte) 0x01;
			SendBuff[4] = (byte) 0x20;
			SendBuff[5] = (byte) 0x23;
			SendBuff[6] = (byte) 0x11;
			SendBuff[7] = (byte) 0x04;
			SendBuff[8] = (byte) 0x10;
			
			SendLen = 9;
			RecvLen[0] = 255;
			
			retCode = acr122.jACR122_DirectTransmit(hReader, SendBuff, SendLen, RecvBuff, RecvLen);
			
			if(retCode == 0)
			{
				if(RecvLen[0] > 3)
				{
					switch(RecvBuff[8])
					{
					case (byte) 0x18: PrintText = "Mifare 4K"; break;
					case (byte) 0x00: PrintText = "Mifare Ultralight";  break;
					case (byte) 0x28: PrintText = "ISO 14443-4 Type A"; break;
					case (byte) 0x08: PrintText = "Mifare 1K"; break;
					case (byte) 0x09: PrintText = "Mifare Mini"; break;
					case (byte) 0x20: PrintText = "Mifare DesFire"; break;
					case (byte) 0x98: PrintText = "GemPlus MPCOS"; break;
					default:
							switch(RecvBuff[3])
							{
							case (byte) 0x23: PrintText = "ISO 14443-4 Type B"; break;
							case (byte) 0x11: PrintText = "FeliCa 212K"; break;
							case (byte) 0x04: PrintText = "Topaz"; break;
							default:
								PrintText = "Unknown contactless card"; 
							}
					}
					
					lCardType.setText(PrintText);
				}
				else
				{
					lCardType.setText("");
				}
							
			}
    		
    	}
    	
    	if(bQuit == e.getSource())
		{
			if(connActive)
			{
				acr122.jACR122_Close(hReader);
			}
			
			this.dispose();
			
		}
    	
    }
    
    
    
    public static void main(String args[]) {
        EventQueue.invokeLater(new Runnable() {
            public void run() {
                new Polling().setVisible(true);
            }
        });
    }
	

}
